source/common/upstream/subset_lb.cc

```diff
@@ -101,6 +112,30 @@ SubsetLoadBalancer::~SubsetLoadBalancer() {
   });
 }
 
+void SubsetLoadBalancer::rebuildSingle() {
+  if (single_) {
+    // Because PriorityUpdateCb doesn't give a modified list (only added and removed), it is
```

Comment:
It does provide the priority that changed, so there is some opportunity for optimizing the case where there are a lot of hosts split between several priorities. Probably not worth it for the initial implementation though